pkgs/build-support/cc-wrapper/add-hardening

```diff
@@ -0,0 +1,61 @@
+hardeningFlags=(fortify stackprotector pic strictoverflow format relro bindnow)
+hardeningFlags+=("${hardeningEnable[@]}")
+hardeningCFlags=()
+hardeningLDFlags=()
+hardeningDisable=${hardeningDisable:-""}
+
+if [[ "$($LD -z 2>&1)" =~ "unknown option" ]]; then
```

Comment:
Given that $LD is constant, it should be possible to do this check at stdenv build time.